spec/foxsi2_count_spectrum.pro

```diff
@@ -89,10 +112,28 @@ counts_coarse = interpol(counts, emid, emid_coarse)	; units still in counts / se
 counts_coarse = counts_coarse*time		; Now units are counts / keV
 y_err_raw = sqrt(counts_coarse)				; units also in counts / keV
 
+;finding counts at mid point of each E bin using average of each E bin
+if keyword_set(averagects) then begin
+    counts_avg_coarse=dblarr(n_elements(emid_coarse))
+    for avg_ct=1,n_elements(emid_coarse)-1  do $
+	counts_avg_coarse[avg_ct-1] = $
+	mean(counts[where((emid ge e1_coarse[avg_ct-1]) and $
+	(emid le e1_coarse[avg_ct]))])	; units still in counts / sec / keV
+    counts_avg_coarse = counts_avg_coarse*time		; Now units are counts / keV
+    y_err_avg_raw = sqrt(counts_avg_coarse)
+endif
```

Comment:
If you include this option in the code (averagects but no smear), you should also include a way to save this information in "result".